v4l2/v4l2_codecbase.cpp

```diff
@@ -731,4 +738,60 @@ bool V4l2CodecBase::mapVideoFrames(int32_t width, int32_t height)
     }
     return true;
 }
+#elif __ENABLE_WAYLAND__
+inline bool V4l2CodecBase::setVaDisplay()
+{
+    VAStatus status;
+    int major, minor;
+    m_vaDisplay = vaGetDisplayWl(m_wlDisplay);
+    status = vaInitialize(m_vaDisplay, &major, &minor);
+    if (status != VA_STATUS_SUCCESS)
+        return false;
+    return true;
+}
+
+inline bool V4l2CodecBase::createVpp()
+{
+    NativeDisplay nativeDisplay;
+    nativeDisplay.type = NATIVE_DISPLAY_VA;
+    nativeDisplay.handle = (intptr_t)m_vaDisplay;
+    m_vpp.reset(createVideoPostProcess(YAMI_VPP_SCALER), releaseVideoPostProcess);
+    return m_vpp->setNativeDisplay(nativeDisplay) == YAMI_SUCCESS;
```

Comment:
is it possible to create wl_buffer from the decoder vaSurface directly?
the effort should be less than it on android.